src/pages/suggestions/new/index.tsx

```diff
@@ -1,92 +1,101 @@
-import React, { useEffect, useState } from "react"
-import { XMarkIcon } from "@heroicons/react/24/outline"
-import ProgressBar from "@/components/new-suggestion/progress-bar/progress-bar"
-import { useRouter } from "next/router"
-import SongInformationArea from "@/components/new-suggestion/areas/song-information.area"
-import ReviewArea from "@/components/new-suggestion/areas/review.area"
-import { Area } from "@/constants/area"
-import { useSelector } from "react-redux"
+import React from "react"
+import { useDispatch, useSelector } from "react-redux"
 import { AppState } from "@/redux/store"
-import { FormProvider, useForm } from "react-hook-form"
-import { InputsSongInformation } from "@/interfaces/new-suggestion"
-import InstrumentsArea from "@/components/new-suggestion/areas/instruments/instruments.area"
+import SuggestionPageSection from "@/components/new-suggestion/suggestion-page-section"
+import { Area } from "@/constants/area"
+import {
+  updateNewSuggestion,
+  initialState,
+  setActiveArea,
+} from "@/redux/slices/new-suggestion.slice"
+import { insertSuggestion, insertSuggestionInstruments } from "@/services/suggestion.service"
 import { Database } from "@/types/database"
-import { Instrument } from "@/types/database-types"
-import { useSupabaseClient } from "@supabase/auth-helpers-react"
-import { getInstruments } from "@/services/instrument.service"
-import Spinner from "@/components/utils/spinner"
-import ErrorPopup from "@/components/popups/error-popup"
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react"
+import { useRouter } from "next/router"
+import { mapInstruments } from "@/helpers/new-suggestion.helper"
+import { InputsSongInformation, NewSuggestionInstrument } from "@/interfaces/new-suggestion"
 
 const NewSuggestion = () => {
-  const router = useRouter()
-  const activeArea = useSelector((state: AppState) => state.newSuggestion.activeArea)
   const suggestion = useSelector((state: AppState) => state.newSuggestion.suggestion)
+  const activeArea = useSelector((state: AppState) => state.newSuggestion.activeArea)
 
-  const methods = useForm<InputsSongInformation>({
-    defaultValues: { ...suggestion, artist: suggestion.artist.join(",") } as InputsSongInformation,
-    shouldFocusError: false,
-  })
+  const supabase = useSupabaseClient<Database>()
+  const dispatch = useDispatch()
+  const router = useRouter()
+  const user = useUser()
 
-  const supabaseClient = useSupabaseClient<Database>()
-  const [instrumentList, setInstrumentList] = useState<Instrument[]>([])
-  const [showSpinner, setShowSpinner] = useState<boolean>(false)
-  const [showLoadingError, setShowLoadingError] = useState<boolean>(false)
+  const saveSuggestion = (onSuccess: () => void, onError: () => void) => {
+    if (user) {
```

Comment:
```suggestion
    if (!user) return
```
less nesting is more readable